CHANGELOG.rst

```diff
@@ -7,6 +7,14 @@
 Change log
 ==========
 
+(unreleased)
+-------------------
+
+**Other changes**
+
+- Imporved quadratic complexity of the naming algorithm in the build process, by keeping track of how many times a base name was used in current and all containing scopes.
```

Comment:
The changelog is meant for users who don't necessarily care about the internals; only about how it affects them. We always update the unreleased version number with each PR according to Semver. Setting the correct version number at the time of the release can get necessarily annoying. 


```suggestion
0.12.1 (unreleased)
-------------------

**Other changes**

- The node-naming algorithm now has constant rather than quadratic time complexity.
```